Updating python support versions

matthewshaver opened the pull request:
## What are you changing in this pull request and why?

Updating the python version support FAQ with the most recent info.

## Checklist
<!--
Uncomment when publishing docs for a prerelease version of dbt:
- [ ] Add versioning components, as described in [Versioning Docs](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-entire-pages)
- [ ] Add a note to the prerelease version [Migration Guide](https://github.com/dbt-labs/docs.getdbt.com/tree/current/website/docs/docs/dbt-versions/core-upgrade)
-->
- [ ] Review the [Content style guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md) so my content adheres to these guidelines.
- [ ] For [docs versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning), review how to [version a whole page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version) and [version a block of content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links that point to deleted pages

runleonarun committed:
Update install-python-compatibility.md

runleonarun committed:
Update install-python-compatibility.md

runleonarun committed:
Merge branch 'current' into runleonarun-patch-15

runleonarun committed:
Update website/docs/faqs/Core/install-python-compatibility.md

matthewshaver committed:
Updating python support versions

matthewshaver commented on website/docs/faqs/Core/install-python-compatibility.md:
```diff
@@ -5,26 +5,12 @@ sidebar_label: 'Python version'
 ---
 
-<VersionBlock firstVersion="1.6">
+Adapter plugins and their dependencies are not always compatible with the latest version of Python. For example, dbt-snowflake v0.15 is not compatible with Python 3.12, but dbt-snowflake versions 0.20+ are.
```
@aranke Is there a more recent example we can provide here?

runleonarun commented:
@matthewshaver Can you fold your feedback into[ this PR](https://github.com/dbt-labs/docs.getdbt.com/pull/5012)?

matthewshaver committed:
Merge branch 'python-versions' into runleonarun-patch-15

matthewshaver committed:
Update install-python-compatibility.md (#5012)

Closes https://github.com/dbt-labs/docs.getdbt.com/issues/5006

## What are you changing in this pull request and why?

Adding an easier to glance at table for Python <> dbt Core version
compatibility.
## Checklist
- [x] Review the [Content style
guide](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/content-style-guide.md)
so my content adheres to these guidelines.
- [ ] For [docs
versioning](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#about-versioning),
review how to [version a whole
page](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#adding-a-new-version)
and [version a block of
content](https://github.com/dbt-labs/docs.getdbt.com/blob/current/contributing/single-sourcing-content.md#versioning-blocks-of-content).
- [ ] Add a checklist item for anything that needs to happen before this
PR is merged, such as "needs technical review" or "change base branch."

Adding or removing pages (delete if not applicable):
- [ ] Add/remove page in `website/sidebars.js`
- [ ] Provide a unique filename for new pages
- [ ] Add an entry for deleted pages in `website/vercel.json`
- [ ] Run link testing locally with `npm run build` to update the links
that point to deleted pages

nghi-ly commented on website/docs/faqs/Core/install-python-compatibility.md:
```diff
@@ -5,26 +5,11 @@ sidebar_label: 'Python version'
+import Pythonmatrix from '/snippets/_python-compatibility-matrix.md';
 
-The latest version of `dbt-core` is compatible with Python versions 3.8, 3.9, 3.10 and 3.11.
+Use this table to match dbt-core versions with their compatible Python versions. New dbt minor versions will add support for new Python3 minor versions as soon as all dependencies can support it. In addition, dbt minor versions will drop support for old Python3 minor versions right before they reach [end of life](https://endoflife.date/python).
```
suggs: for conciseness and slightly more translation friendly

```suggestion
Use this table to match dbt-core versions with their compatible Python versions. New [dbt minor versions](/docs/dbt-versions/core#minor-versions) will add support for new Python3 minor versions when all dependencies can support it. In addition, dbt minor versions will withdraw support for old Python3 minor versions before their [end of life](https://endoflife.date/python).
```

nghi-ly reviewed (approved):
looks great! left one non-blocking sugg

matthewshaver committed:
Update website/docs/faqs/Core/install-python-compatibility.md

Co-authored-by: Ly Nguyen <107218380+nghi-ly@users.noreply.github.com>
